backend/framework/qlf/dashboard/bokeh/qaskyresid/main.py

```diff
@@ -172,34 +208,111 @@ def load_qa(self):
+
+
+        #hdul=fits.open('/app/spectro/redux/exposures/20180701/00000005/sframe-b0-00000005.fits')
+        try:
+            hdul = load_fits(self.selected_process_id, cam)
+            wlength = hdul['WAVELENGTH'].data
+        except Exception as err:
+            logger.info('{}'.format(err))   
+
+
+        p_s = []
+        # for obj in obj_list:
+        available_obj = set(fmap[1].data['OBJTYPE'])
+        # if 'SKY' in available_obj:
+        #     available_obj.remove('SKY')
+
+        for obj in available_obj:
+            spec_tooltip = """
+                <div>
+                    <div>
+                        <span style="font-size: 12px; font-weight: bold; color: #303030;">Wavelength: </span>
+                        <span style="font-size: 13px; color: #515151">@wlength &#8491</span>
+                    </div>
+                    <div>
+                        <span style="font-size: 12px; font-weight: bold; color: #303030;">Counts: </span>
+                        <span style="font-size: 13px; color: #515151;">@spec</span>
+                    </div>
+                </div>
+            """
+            spec_hover = HoverTool(tooltips=spec_tooltip, mode='vline')
+
+            try:
+                if obj == 'STD':
+                    obj_name = 'STAR'
+                else:
+                    obj_name = obj
+                arg = np.random.choice(gen_info[obj_name+'_FIBERID'],1)
+                spec = hdul['FLUX'].data[arg][0]
+
+                spec_source = ColumnDataSource(data={
+                'wlength': wlength,
+                'spec': spec,
+                })
+                p_spec = Figure(title= obj_name+' Spectrum Sample (Fiber ID: %d)'%arg, 
+                    x_axis_label='Wavelength (A)', y_axis_label=obj_name+' Flux (counts)',
+                    plot_width = 500, plot_height = 240,
+                    tools=[spec_hover,"pan,box_zoom,reset,crosshair, lasso_select" ])
+
+                p_spec.line('wlength', 'spec', source = spec_source)
+                p_s.append(p_spec)
+            except Exception as err:
+                sys.exit(err)
+
+
+        width_tb, height_tb = 400, 140
+
+        alert = alert_table(nrg, wrg)
+        tb_alert = Div(text=alert, width=width_tb, height=height_tb)
+
+        info = metric_table('Sky Residuals', 'comments', 'keyname') #, curexp=check_spectra['MED_RESID'], refexp=check_spectra['MED_RESID_REF'])
+        tb_metric =Div(text=info, width=width_tb, height=height_tb)
+
+
         # p2txt = column(widgetbox(info_col), row(column(p1, p2), tbinfo), fiber_hist  )
-        layout = column(widgetbox(info_col, css_classes=["header"]),
-                      widgetbox(tbinfo, css_classes=["table-ranges"]),
-                      p1,
-                      fiber_hist,
-                      p2,
-                      css_classes=["display-grid-skyresid"])
+        try:
+            # layout = column(widgetbox(info_col, css_classes=["header"]),
```

Comment:
If you are not going to use it just delete it. We can recover code from git history